crates/router/src/core/user/theme.rs

```diff
@@ -0,0 +1,212 @@
+use api_models::user::theme as theme_api;
+use common_utils::{ext_traits::Encode, types::theme::ThemeLineage};
+use diesel_models::user::theme::ThemeNew;
+use error_stack::ResultExt;
+use hyperswitch_domain_models::api::ApplicationResponse;
+use uuid::Uuid;
+
+use crate::{
+    core::errors::{StorageErrorExt, UserErrors, UserResponse},
+    routes::SessionState,
+    utils::user::theme as theme_utils,
+};
+
+pub async fn get_theme_using_lineage(
+    state: SessionState,
+    lineage: ThemeLineage,
+) -> UserResponse<theme_api::GetThemeResponse> {
+    let theme = state
+        .global_store
+        .find_theme_by_lineage(lineage)
+        .await
+        .to_not_found_response(UserErrors::ThemeNotFound)?;
+
+    let file = theme_utils::retrieve_file_from_theme_bucket(
+        &state,
+        &theme_utils::get_theme_file_key(&theme.theme_id),
+    )
+    .await?;
+
+    let parsed_data =
+        serde_json::from_slice(&file).change_context(UserErrors::InternalServerError)?;
```

Comment:
In the data parsing case, we can't log the the data from S3 as it will be in bytes and we won't be able to understand it from the logs.

I wouldn't like to add the log because it won't give us any more information that the the error itself.

I changed in other cases though.